src/core/test/base/src/main/scala/TestBase.scala

```diff
@@ -163,11 +175,16 @@ abstract class TestBase extends FunSuite with BeforeAndAfterEachTestData with Be
   }
 
   def time[R](block: => R): R = {
+    val (result, t) = getTime(block)
+    println(s"Elapsed time: ${t / 1e9} sec")
+    result
+  }
+
+  def getTime[R](block: => R): (R, Long) = {
     val t0     = System.nanoTime()
     val result = block
     val t1     = System.nanoTime()
-    println(s"Elapsed time: ${(t1 - t0) / 1e9} sec")
-    result
+    (result, t1-t0)
```

Comment:
This looks very similar to the `timeit` function in Python - you might want to emulate that logic here. See this paper that talks about how to measure performance in MATLAB that inspired the whole thing: http://www.mathworks.com/matlabcentral/fileexchange/18510-matlab-performance-measurement